daemon/endpoint.go

```diff
@@ -211,6 +211,44 @@ func (h *putEndpointID) Handle(params PutEndpointIDParams) middleware.Responder
 	if err != nil {
 		apierror.Error(code, err)
 	}
+
+	// Wait for endpoint to be ready if specified in API call.
+	if params.Endpoint.SyncBuildEndpoint {
```

Comment:
Quick question. Please correct me if I am wrong. Will this be a problem if we try to create a docker container without labels. Since in that case, it will be EndpointStateCreating /StateWaitingForIdentity(Not sure how this behavior has changed with the new Init identity)? I believe, in docker container creation, there is a transition from EndpointStateCreating to other states which is spilt between in PUT (creation) and PATCH. Will that be a problem, now that we are synchronously waiting for the state to be `ready` in PUT itself?


This is fine case of K8s which does not have the split between PUT/PATCH but here, will we be

Reply:
The flag is only enabled via CNI so it will not affect libnetwork setups